ndk/src/surface_texture.rs

```diff
@@ -0,0 +1,88 @@
+//! Bindings for [`ffi::ASurfaceTexture`]
+use crate::native_window::NativeWindow;
+use jni_sys::jobject;
+use jni_sys::JNIEnv;
+use std::ptr::NonNull;
+
+#[derive(Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+pub struct SurfaceTexture {
+    ptr: NonNull<ffi::ASurfaceTexture>,
+}
+
+unsafe impl Send for SurfaceTexture {}
+
+#[derive(Debug)]
+pub struct PosixError(pub i32);
```

Comment:
This type seems pretty useless, should it at least implement `Error` and `Display`?

I wonder if there already exists a helper for it _somewhere_ though, nicely stringifying everything based on `errno-base.h`. I thought we already had such a thing somewhere in this crate but I might be mistaken.